### DISCUSSION TITLE
Invalidate agressive cache

### DISCUSSION THREAD

**Discussion by SbiCA** [ravendb/ravendb#18480] [Q&A] [answered]
Is there a way to invalidate / evict the aggressive caches?

I have a scenario where I need to invalidate it occasionally (once a week) and I wanted to make sure if that chance occurs that we don't serve stale data hence I'd like to add a functionality to invalidate said caches. 

Atm. we're using aggressive cache on session level with defaults
``` c# 
using var session = _documentStore.OpenAsyncSession();
await session.Advanced.DocumentStore.AggressivelyCacheAsync();
var list = await session.Query<TenantConfiguration>().ToListAsync();
```

Thx for your efforts!

**Comment by SbiCA** [comment]
I think I found the answer myself btw. would be good if someone could confirm, if that's the way to properly invalidate the caches? @ayende 

```c#
 using var session = _documentStore.OpenAsyncSession();
 session.Advanced.RequestExecutor.Cache.Clear();
 ```

**Reply by ayende** [reply, marked as answer]
It would be cheaper to do:

```
Interlocked.Increment(ref store.GetRequestExecutor().Cache.Generation);
```

This will not clear the cache, but force a refresh from the server for all cached requests.
That way, if the data didn't change, you'll still get `304` reply

**Reply by SbiCA** [reply]
Thanks can confirm it work!
